app/controllers/api/v1/digital_objects_controller.rb

```diff
@@ -24,14 +24,14 @@ def show
       # POST /digital_objects.json
       def create
         digital_object_data = create_or_update_params
-        digital_object_record = DigitalObjectRecord.create
-        @digital_object = DigitalObject::Base.from_serialized_form(digital_object_record, digital_object_data)
+        @digital_object = Hyacinth.config.digital_object_types.key_to_class(digital_object_data['digital_object_type']).new
+        @digital_object.set_digital_object_data(digital_object_data, true)
```

Comment:
if we start thinking of a DigitalObject instance as, essentially, a decorated deeply nested hash - is the set_digital_object_data method essentially merge or replace? Could we name the methods like the corresponding Hash methods?

Reply:
@barmintor It's sort of a merge/replace, but there's also some data transformation / augmentation involved for some fields.  Sometimes a string key is provided for a field and it's resolved to an object like a Project.  Other times, for a URI dynamic field, a minimal hash is provided for a field value and that hash will have other key/value pairs added to it (pref_label, authority).  There are also some params expected by set_digital_object_data that don't map 1:1 with DigitalObject field names (e.g. publish_to, unpublish_from), but they do end up setting other fields (e.g. publish_target_entries).  Does that change the way you feel about it?

Reply:
`merge` does seem a lot clearer to me even though it is doing some data transformation. 

I know this is probably only a Carla problem, but when I read `set_digital_object_data` I always have to do a double take and make sure it isn't `set_dynamic_field_data`. `merge` implies that it accepts and merges all the data associated with a digital object.

Reply:
I want to merge this pull request. What should we do about this? Should we make a ticket to change the name of this method, or just leave as is?

Reply:
I'm open to other method names if you want something shorter, but the other case I'll make against `merge` is based on the method signature:

`def set_digital_object_data(new_digital_object_data, merge_dynamic_fields)`

If we called it `merge`, then we'd be optionally passing `merge` a second "but don't merge everything" arg.  Seems weird.

Reply:
@cgalarza I'd say no harm in merging this PR, and we can break this discussion out into a separate conversation.  The change isn't related to the other changes in your PR.

Reply:
Okay, let's do that!